mk/config.mk

```diff
@@ -244,10 +244,27 @@ CFG_CORE_SANITIZE_UNDEFINED ?= n
 CFG_CORE_SANITIZE_KADDRESS ?= n
 
 # Device Tree support
-# When enabled, the TEE _start function expects to find the address of a
-# Device Tree Blob (DTB) in register r2. The DT parsing code relies on
-# libfdt.  Currently only used to add the optee node and a reserved-memory
-# node for shared memory.
+#
+# When CFG_DT is enabledn core embeds the FDT library (libfdt) allowing DTB
+# image parsing from the core.
+#
+# When CFG_EMBEDDED_SECURE_DT is enabled, TEE core embeds a statically linked
+# device tree blob in a read-only pageable section of the core.
+#
+# When CFG_SECURE_DTS is defined, it is expected to provide the list of the base
+# name of the device tree source file for which a device tree blob is generated
+# when core is built.
```

Comment:
What happens if `CFG_SECURE_DTS` contains several files? Are they all embedded in the TEE binary? How are they consolidated?

Reply:
If `CFG_SECURE_DTS` list several files, a DTB is generated for each of them.

`CFG_EMBEDDED_SECURE_DT=y`, **core/sub.mk** verifies that `CFG_SECURE_DTS` contains a single entry.

Reply:
> If `CFG_SECURE_DTS` list several files, a DTB is generated for each of them.

Good.

> `CFG_EMBEDDED_SECURE_DT=y`, **core/sub.mk** verifies that `CFG_SECURE_DTS` contains a single entry.

It would be more convenient if `CFG_EMBEDDED_SECURE_DT` would be the name of that file. And call it `CFG_EMBEDDED_SECURE_DTS`. Then there is no error case to test, and only one config variable needs to be set.

Reply:
So `CFG_SECURE_DTS`  and `CFG_EMBEDDED_SECURE_DTS` would be exclusive. Is that fine to you?

Looks like `CFG_SECURE_DTS` listing several files is useless. Do you think it is still accurate to introduce `CFG_SECURE_DTS` if it is never used?

Reply:
I suppose we could drop `CFG_SECURE_DTS` and keep only `CFG_EMBEDDED_SECURE_DTS`. In other words, make things as simple as possible.

Reply:
Ok. let's make it simple.
I will keep however the currently proposed secure_dt.mk makefile script so that it supports generating several DTBs, even if we don't use it for now.

Reply:
Not a good idea IMO. "Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away. Antoine de Saint-Exupery".

Reply:
OK